src/main/java/controllers/SceneManager.java

```diff
@@ -0,0 +1,78 @@
+package controllers;
+
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import views.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Singleton class for managing scenes.
+ */
+public class SceneManager {
+
+    /**
+     * Singleton instance.
+     */
+    private static final SceneManager INSTANCE = new SceneManager();
+
+    private final Scene mainMenuScene = new Scene(new MainMenuPane(), 800 / 2.0, 600);
+
+    private final Scene leagueSimulatorScene = new Scene(new LeagueSimulatorPane(), 800, 600);
+
+    private final Scene leaderboardScene = new Scene(new LeaderboardPane(), 800 / 2.0, 600);
+
+    private final Scene teamEditorScene = new Scene(new TeamEditorPane(), 400, 750);
+
+    private Map<Class<? extends GamePane>, Scene> scenes = new HashMap<Class<? extends GamePane>, Scene>(){{
+        put(MainMenuPane.class, mainMenuScene);
+        put(LeagueSimulatorPane.class, leagueSimulatorScene);
+        put(TeamEditorPane.class, teamEditorScene);
+        put(LeaderboardPane.class, leaderboardScene);
+
+    }};
+
+    /**
+     * Primary stage.
+     */
+    private Stage stage;
+
+    public void setStage(final Stage stage) {
+        if (this.stage != null) {
+            throw new IllegalStateException("Primary stage is already initialized!");
+        }
+
+        this.stage = stage;
+    }
+
+
+    private void showScene(final Scene scene) {
+        if (stage == null) {
+            return;
+        }
+
+        stage.hide();
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    /**
+     * Replaces the current {@link GamePane} with another one.
+     *
+     * @param pane New pane to display.
+     */
+    public void showPane(final Class<? extends GamePane> pane) {
+        // TODO
```

Comment:
I would suggest removing the TODO if it's no longer necessary :)

Reply:
Also agree